example/test/react_test_components.dart

```diff
@@ -223,23 +223,168 @@ class _ContextConsumerComponent extends react.Component {
   }
 }
 
-var contextConsumerComponent = react.registerComponent(() => new _ContextConsumerComponent());
+var legacyContextConsumerComponent = react.registerComponent(() => new _LegacyContextConsumerComponent());
 
-class _GrandchildContextConsumerComponent extends react.Component {
+class _GrandchildLegacyContextConsumerComponent extends react.Component {
   @override
   Iterable<String> get contextKeys => const ['renderCount'];
 
   render() {
     return react.ul({
       'key': 'ul'
     }, [
-      'GrandchildContextConsumerComponent.context: ',
+      'LegacyGrandchildContextConsumerComponent.context: ',
       context.toString(),
     ]);
   }
 }
 
-var grandchildContextConsumerComponent = react.registerComponent(() => new _GrandchildContextConsumerComponent());
+var grandchildLegacyContextConsumerComponent =
+    react.registerComponent(() => new _GrandchildLegacyContextConsumerComponent());
+
+////
+// REACT NEW CONTEXT COMPONENTS
+////
+class _NewContextRefComponent extends react.Component2 {
+  render() {
+    return react.div({}, props['children']);
+  }
+
+  test() {
+    print('test');
+  }
+}
+
+var newContextRefComponent = react.registerComponent(() => new _NewContextRefComponent());
+
+int calculateChangedBits(currentValue, nextValue) {
+  int result = 1 << 1;
+  if (nextValue['renderCount'] % 2 == 0) {
+    result |= 1 << 2;
+  }
+  return result;
```

Comment:
I have literally no idea what is happening, or why it is happening.  The second argument of `createContext` deserves some documentation explaining what it does.  For instance, why does it always return an integer?

Reply:
The React team doesn't even have documentation around it. They act as if it doesn't exist even though it obviously does, i think this may still be a WIP/secret or experimental feature...

The idea is that you return a bit that indicates which parts/or how the context value changed, then you can use the observedBits props on a consumer and they will only re-render if the bits they are observing are returned from calculateChangedBits

Reply:
Sorry for the crazy delayed reply here @kealjones-wk - a doc comment about how you perceive it to work (like what you just typed above) would be a great addition!